### PR TITLE
Pr code metrics

### PR DESCRIPTION
**Pull request by aubreyquinn**
## Summary

Added some unit tests to code metrics

## Test Plan

need to add more unit tests

### CODE REVIEW

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/codeMetrics.ts [hunk 1]
```diff
@@ -29,19 +27,72 @@ class CodeMetrics {
...
+    return this._metrics
+  }
+
+  public set metrics (newMetrics: Metrics) {
+    // throw error if input is incorrect
+    this.metrics = newMetrics
+    this.setSufficientTestCode()
+  }
+
+  public get size (): string {
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/codeMetrics.ts:49]: Most of these should be set internally. You shouldn't need to expose them for the unit tests, so please don't add them

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/codeMetrics.ts [hunk 2]
```diff
@@ -50,22 +101,16 @@ class CodeMetrics {
     return indicator
   }
 
-  public get isSmall (): boolean {
+  public isSmall (): boolean {
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/codeMetrics.ts:104]: Can you switch this back to a setter?

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/iMetrics.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/updaters/iMetrics.ts:1]: You shouldn't need this now

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/wrappers/ProcessWrapper.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/wrappers/ProcessWrapper.ts:1]: You shouldn't need this now either

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import { instance, mock, verify, when } from 'ts-mockito'
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts:1]: Missing copyright

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+describe('codeMetrics.ts', (): void => {
+  let execSyncResult: IExecSyncResult
+  let taskLibWrapper: TaskLibWrapper
+  let consoleWrapper: ConsoleWrapper
+
+  beforeEach((): void => {
+    process.env.SYSTEM_PULLREQUEST_TARGETBRANCH = 'refs/heads/develop'
+    process.env.SYSTEM_PULLREQUEST_PULLREQUESTID = '12345'
+
+    execSyncResult = new ExecSyncResult()
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts:20]: I don't think you need the environment variables or execsyncresult here

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    execSyncResult.stdout = '1\t2\tFile.txt'
+
+    taskLibWrapper = mock(TaskLibWrapper)
+    consoleWrapper = mock(consoleWrapper)
+    when(taskLibWrapper.execSync('git', 'diff --numstat origin/develop...pull/12345/merge')).thenReturn(execSyncResult)
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts:25]: Shouldn't need this when

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  })
+
+  afterEach((): void => {
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/codeMetrics.spec.ts:28]: Shouldn't need the afterEach

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/pullRequestComments.spec.ts [hunk 9]
```diff
@@ -1,15 +1,16 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import async from 'async'
-import os from 'os'
+import { instance, mock, verify, when } from 'ts-mockito'
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/pullRequestComments.spec.ts:4]: Is there a reason for reordering this?

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/pullRequestComments.spec.ts [hunk 10]
```diff
@@ -18,8 +19,8 @@ describe('pullRequestComments.ts', (): void => {
   beforeEach((): void => {
     codeMetrics = mock(CodeMetrics)
     when(codeMetrics.baseSize).thenReturn(250)
-    when(codeMetrics.hasSufficientTestCode).thenReturn(true)
-    when(codeMetrics.isSmall).thenReturn(true)
+    // when(codeMetrics.sufficientTestCode).thenReturn(true)
```
Comment (muiriswoulfe) [inline PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/pullRequestComments.spec.ts:22]: Please uncomment this code

**Review by muiriswoulfe — CHANGES_REQUESTED** [review]

File: PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/pullRequestComments.spec.ts (hunk 9 above)
Comment (aubreyquinn) [reply PipelinesTasks/PRMetrics/buildAndReleaseTask/tests/updaters/pullRequestComments.spec.ts:4]: No

**Review by muiriswoulfe — APPROVED** [review]

**Pull request merged**